packages/in_app_purchase/example/lib/main.dart

```diff
@@ -0,0 +1,372 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: avoid_print
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:in_app_purchase/in_app_purchase.dart';
+import 'package:in_app_purchase_tizen/billing_manager_wrappers.dart';
+import 'package:in_app_purchase_tizen/in_app_purchase_tizen.dart';
+
+void main() {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  InAppPurchaseTizenPlatform.registerPlatform();
+  runApp(_MyApp());
+}
+
+// To try without auto-consume, change `true` to `false` here.
+const bool _kAutoConsume = true;
+
+/// The [_kConsumableType]'s value:
+/// "1": CONSUMABLE
+/// "2": NON-CONSUMABLE
+/// "3": LIMITED-PERIOD
+/// "4": SUBSCRIPTION
+const int _kConsumableType = 1;
+const String _kAppId = '3201504002021';
+const String _kCustumId = '810000047372';
+const String _kServerType = 'DEV';
+const String _kCountryCode = 'US';
+const String _kItemType = '2';
+const String _kPageSize = '20';
+const String _kPageNum = '1';
+// ignore: unused_element
+const String _kSecurityKey = '7db0YeFxHPtajwvVShBC5auI4iSJraKy8yhPZe13yvM=';
+
+/// The check value is used by the DPI service to verify Purchase List and Products List API requests.
+/// It is a Base64 hash generated by applying the HMAC SHA256 algorithm on a concatenated string of parameters using the DPI security key.
+///
+/// The [_kProductCheckValue] is generated with [_kAppId] and [_kCountryCode] and [_kSecurityKey];
+/// var hmacSha256 = Hmac(sha256, utf8.encode(_kSecurityKey));
+/// String _kProductCheckValue = base64Url.encode(hmacSha256.convert(utf8.encode(_kAppId + _kCountryCode)).bytes);
+String _kProductCheckValue = 'sbKzQJrFdeLkfXTQUdjjq4jt7ykx79QmJno9ZLam+SM=';
+
+/// The [_kPurchaseCheckValue] is generated with [_kAppId] and [_kCustumId] and [_kCountryCode] and [_kItemType] and [_kSecurityKey];
+/// var hmacSha256 = Hmac(sha256, utf8.encode(_kSecurityKey));
+/// String _kPurchaseCheckValue = base64Url.encode(hmacSha256.convert(utf8.encode(_kAppId + _kCustumId + _kCountryCode + _kItemType)).bytes);
+String _kPurchaseCheckValue = 'eAKHVR/FUfgEmDb3KmtgkozWAKyEb1OWXKdvkUV4km0=';
+
+// WARNING: Don't change order of `_kRequestParams`.
+List<String> _kRequestParams = <String>[
+  _kAppId,
+  _kCountryCode,
+  _kItemType,
+  _kPageSize,
+  _kPageNum,
+  _kServerType,
+  _kProductCheckValue,
+  _kPurchaseCheckValue,
```

Comment:
Should all these values be provided by the user? Can't they be calculated automatically or have default values? The API will not be very usable if all this information has to be provided by the user. Also `queryProductDetails` basically expects a `Set` of identifiers, not a list of params as I mentioned above.

Reply:
Generate `_kProductCheckValue` and `_kPurchaseCheckValue` need `_kAppId` and `_kSecurityKey`, which are both provided by user(in [Samsung Checkout DPI portal](https://dpi.samsungcheckout.com/))

Reply:
Are the check values expected to be calculated in advance and hard-coded into the code (for real world apps)? Should `kSecurityKey` not be present in the user code for security reasons? (I can't find guidance from the API references.)

If not, you could provide utility functions that accept the required parameters + `kSecurityKey` and return the check values. (Preferably in the plugin API itself or otherwise in the example app code.)

Reply:
OK, I find the [documentation](https://developer.samsung.com/smarttv/develop/tizen-net-tv/guides/samsung-checkout/implementing-the-purchase-process.html#Generating-Check-Values), said generate check value during runtime, and not expose security key. 
But how can I not expose `kSecurityKey`? It is provided from user.

Reply:
I don't know. The document says "You can use a key management server" so if the user wants to generate the check value at runtime, they could retrieve the key online. Otherwise the key (or the check value) should be hard-coded into the code as far as I can see.

Reply:
fixed.
Check value will be generated at runtime.